criu/mount.c

```diff
@@ -3585,6 +3585,10 @@ int mntns_get_root_fd(struct ns_id *mntns)
 {
 	if (!(root_ns_mask & CLONE_NEWNS))
 		return __mntns_get_root_fd(0);
+
+	if (!mntns)
```

Comment:
Do we know where can it be called with NULL? Or it is just a precaution?

Reply:
```
3632int mntns_get_root_by_mnt_id(int mnt_id)
3633{
    1. assign_zero: Assigning: mntns = NULL.
3634        struct ns_id *mntns = NULL;
3635
    2. Condition root_ns_mask & 131072, taking false branch.
3636        if (root_ns_mask & CLONE_NEWNS) {
3637                mntns = lookup_nsid_by_mnt_id(mnt_id);
3638                BUG_ON(mntns == NULL);
3639        }
3640
    CID 181217 (#1 of 1): Explicit null dereferenced (FORWARD_NULL)3. var_deref_model: Passing null pointer mntns to mntns_get_root_fd, which dereferences it. [show details]
3641        return mntns_get_root_fd(mntns);
3642}
```
That is the actual coverity warning. So it looks like it could theoretically be called with `NULL`. Not sure if it will actually happen.

Reply:
In case in mntns_get_root_by_mnt_id we take true branch of `if (root_ns_mask & CLONE_NEWNS)`, when mntns is not null. And in mntns_get_root_fd we have special branch for `if (!(root_ns_mask & CLONE_NEWNS))`:

```
if (!(root_ns_mask & CLONE_NEWNS))
        return __mntns_get_root_fd(0);
```

So it's a false positive.

But probably it is still a good thing to make coverity happy.

Reply:
Sounds good. Thanks for the review. I kind of expected it to be a false positive, but the change to make coverity happy is not that big and also not wrong.